torch_geometric/explain/explainer.py

```diff
@@ -39,11 +44,19 @@ def __init__(
         self.model = model
         self.algorithm = algorithm
 
+        # check if model is heterogeneous
+        self.is_hetero = False
+        for _, module in self.model.named_modules():
+            if isinstance(module, torch.nn.ModuleDict):
```

Comment:
This might be a flaky way of detecting wether a model is heterogenous. Because a user might use a `ModuleDict` for other reasons too. 
Lets set `self.is_hetero` in `__call__` and then call `self.algorihm.support` again if `is_hetero` is `True`.

Reply:
I think I understand what you're saying and pushed a commit, but not 100% sure on if the implementation is what you envisioned.